dateutil/test/test_parser.py

```diff
@@ -917,6 +917,18 @@ def test_pre_12_year_same_month(self):
         dtstr = '0003-03-04'
         assert parse(dtstr) == datetime(3, 3, 4)
 
+    def test_rounding_float_like_strings(self):
```

Comment:
This is a mix of `pytest` and `unittest` style tests. For the `unittest` style tests, you use `self.assertEqual` and the name would be `testRoundingFloatLikeStrings`. I'm fine with you de-indenting it so it's not in a class and use `pytest`, like so:

```python
@pytest.mark.parametrize('dtstr,dt', [
    ('5.6h', datetime(2009, 9, 25, 5, 36)),
    ('5.6m', datetime(2009, 9, 25, 0, 5, 36)),
    ('5.6s', datetime(2009, 9, 25, 0, 0, 5, 600000))
]
def test_rounding_floatlike_strings(dtstr, dt):
    assert parse(dtstr, default=datetime(2003, 9, 25)) == dt
```

Though actually `'5.6s'` never had this problem (that's fine, but it might be worth a comment).